src/Meshes/overlap_mesh.jl

```diff
@@ -0,0 +1,13 @@
+function overlap_mesh(mesh1, mesh2)
+    @assert mesh1.domain == mesh2.domain
+    x1c = x_coords(mesh1) ∪ x_coords(mesh2)
+    x2c = y_coords(mesh1) ∪ y_coords(mesh2)
+    sort!(x1c)
+    sort!(x2c)
+    return Mesh2D(mesh1.domain, x1c, x2c)
+end
+
+x_coords(mesh::EquispacedRectangleMesh) = mesh.range1
+y_coords(mesh::EquispacedRectangleMesh) = mesh.range2
+x_coords(mesh::Mesh2D) = mesh.coordinates[:, 1]
+y_coords(mesh::Mesh2D) = mesh.coordinates[:, 2]
```

Comment:
I'm wondering if it would be better to define the `overlap_mesh` as its own type (i.e., `struct`). That way, you wouldn't have to redefine the `x_coords` and `y_coords`, but you could extract those by dispatching on the underlying mesh?

Reply:
It's not that important, I guess. Your way works as well. I was just commenting on the design that would be a bit different this way, compared to the existing API design we had so far for meshes and topologies. So far, Mesh objects have been just a data structure and we have left the functions to access those data (such as the coordinates field) in Topologies and they were dispatched according to the underlying mesh.